### PR TITLE
KOA-5456 Deprecate old colour tokens

### PR DESCRIPTION
**Pull request by marianeum**
We want to deprecate the old colour tokens. For xml tokens we need to do this as a lint check as it's not possible to deprecate tokens.

+ [x] Check this if you have read and followed the [contributing guidelines](https://github.com/Skyscanner/backpack-android/blob/main/CONTRIBUTING.md)

Remember to include the following changes:
+ [x] Component `README.md`
+ [x] Tests
+ [x] Docs (either update [backpack-docs](https://github.com/Skyscanner/backpack-docs) now, or create a follow up ticket)

If you are curious about how we review, please read through the [code review guidelines](https://github.com/Skyscanner/backpack/blob/main/CODE_REVIEW_GUIDELINES.md)

### CODE REVIEW

File: backpack-lint/src/main/java/net/skyscanner/backpack/lint/check/BpkDeprecatedColorUsageDetector.kt [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+        ISSUE,
+        context.getElementLocation(element),
+        EXPLANATION
+      )
+    }
+  }
+
+  override fun appliesToResourceRefs(): Boolean = true
+
+  override fun visitResourceReference(
```
Comment (marianeum) [inline backpack-lint/src/main/java/net/skyscanner/backpack/lint/check/BpkDeprecatedColorUsageDetector.kt:92]: this covers a colour being used in code - ie `R.color.bpkPrimary`

**Review by frugoman — APPROVED** [review]
Awesome job!

File: backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/tokens/BpkColor.kt [hunk 2]
```diff
@@ -22,80 +22,116 @@
 package net.skyscanner.backpack.compose.tokens
 
 import androidx.compose.ui.graphics.Color
+import kotlin.Deprecated
 
 public object BpkColor {
+  @Deprecated("This colour is now deprecated. Please switch to the new semantic colours - see internal New Colours documentation")
```
Comment (bvitaliyg) [inline backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/tokens/BpkColor.kt:28]: I believe it's possible to use the compile-time constant here instead. Please consider it.

File: backpack-lint/src/main/java/net/skyscanner/backpack/lint/check/BpkDeprecatedColorUsageDetector.kt [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+        ISSUE,
+        context.getValueLocation(attribute),
+        EXPLANATION
+      )
+    }
+  }
+
+  override fun getApplicableElements(): Collection<String> = listOf("color")
+
+  override fun visitElement(context: XmlContext, element: Element) {
```
[the remark this answers is not in the document]
Comment (bvitaliyg) [reply backpack-lint/src/main/java/net/skyscanner/backpack/lint/check/BpkDeprecatedColorUsageDetector.kt:80]: What about colors being used in XML layout – does it cover it as well?

**Review by bvitaliyg — COMMENTED** [review]
Have few questions

File: backpack-lint/src/main/java/net/skyscanner/backpack/lint/check/BpkDeprecatedColorUsageDetector.kt (hunk 3 above)
[the remark this answers is not in the document]
Comment (marianeum) [reply backpack-lint/src/main/java/net/skyscanner/backpack/lint/check/BpkDeprecatedColorUsageDetector.kt:80]: the colour being used as an attribute above covers that, yes

File: backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/tokens/BpkColor.kt (hunk 2 above)
Comment (marianeum) [reply backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/tokens/BpkColor.kt:28]: i'll take a look

**Review by bvitaliyg — APPROVED** [review]

**Comment by bvitaliyg** [comment]
Is there any blockers for the release?

**Pull request merged**